agent/consul/state/catalog.go

```diff
@@ -289,9 +289,13 @@ func validateRegisterRequestPeerNamesTxn(_ ReadTxn, args *structs.RegisterReques
 	}
 
 	{
+		// todo alex -- is this correct?
 		// TODO(peering): validate the node's peering exists (skip check on restore)
-		peerName := args.PeerName
-		peerNames[peerName] = struct{}{}
+		if len(peerNames) == 0 {
+			peerName := args.PeerName
+			peerNames[peerName] = struct{}{}
+		}
```

Comment:
I think the len check isn't necessary here (spending O(1) to save O(1) if my memory serves me right)

Reply:
This change should be reverted because it loosens the validation.

Say that `args.PeerName == "foo"` and `args.Service.PeerName == "bar"`. 

This input would not return an error because `peerNames` would only contain `bar`, despite there being conflicting peer names for a service on a node.

Reply:
This should have been reverted ! 🤦🏼